components/form/TaskForm.tsx

```diff
@@ -98,9 +116,60 @@ export function CreateTaskForm({
                     }
                     testId={"form-field-billable"}
                 />
-                <FromButtons onSubmit={onSubmit} onCancel={onCancel} />
+                <FromButtons
+                    onSubmit={formOnSubmit}
+                    onCancel={abortSubmission}
+                />
             </FormContainer>
             {errorMessage && <FormAlerts errorMessage={errorMessage} />}
+        </FormBackground>
+    )
+}
+
+export const CreateTaskForm = (props: CreateTaskFormProps): JSX.Element => {
+    const { postTask } = useUpdateTasks()
+
+    const [errorMessage, setErrorMessage] = useState<string>("")
+    const errorHandler = (error: Error) => setErrorMessage(`${error}`)
+
+    const onSubmit = async (task: Task) => {
+        const newTask = await postTask(task, errorHandler)
+        return props.afterSubmit && props.afterSubmit(newTask)
+    }
+
+    return (
+        <>
+            <TaskForm {...props} onSubmit={onSubmit} />
+            {errorMessage && (
+                <>
+                    <ErrorAlert />
+                    <Box>{errorMessage}</Box>
+                </>
+            )}
+        </>
+    )
+}
+
+export const EditTaskForm = (props: EditTaskFormProps): JSX.Element => {
+    const { putTask } = useUpdateTasks()
+
+    const [errorMessage, setErrorMessage] = useState<string>("")
+    const errorHandler = (error: Error) => setErrorMessage(`${error}`)
+
+    const onSubmit = async (task: Task) => {
+        const updatedTask = await putTask(task, errorHandler)
+        return props.afterSubmit && props.afterSubmit(updatedTask)
+    }
+
+    return (
+        <>
+            <TaskForm {...props} onSubmit={onSubmit} />
+            {errorMessage && (
+                <>
+                    <ErrorAlert />
+                    <Box>{errorMessage}</Box>
+                </>
```

Comment:
We could probably use the `FormAlerts` component you recently built instead:

https://github.com/three-consulting/epoc-frontend/blob/c91049e326908e32543f2a272f6dce0b841ecc90/components/common/FormFields.tsx#L125-L130